### PR TITLE
Mark lester/1652

### PR DESCRIPTION
**Pull request by mark-lester**
for https://github.com/osm-search/Nominatim/issues/1652

### CODE REVIEW

File: lib/SearchDescription.php [hunk 1]
```diff
@@ -5,6 +5,7 @@
 require_once(CONST_BasePath.'/lib/SpecialSearchOperator.php');
 require_once(CONST_BasePath.'/lib/SearchContext.php');
 require_once(CONST_BasePath.'/lib/Result.php');
+require_once(CONST_BasePath.'/lib/names.php');
```
Comment (mtmail) [inline lib/SearchDescription.php:8]: The one function in the module does so little, I think we don't need a module.

File: lib/SearchDescription.php [hunk 2]
```diff
@@ -522,26 +523,30 @@ private function queryNearbyPoi(&$oDB, $iLimit)
...
+        $sIndexName=$this->poiIndex();
+        $bIndex = $oDB->indexExists($sIndexName);
 
-        if ($oDB->tableExists($sPoiTable)) {
-            $sSQL = 'SELECT place_id FROM '.$sPoiTable.' ct';
-            if ($this->oContext->sqlCountryList) {
-                $sSQL .= ' JOIN placex USING (place_id)';
-            }
+        if ($bIndex) {
+            $qstart=false;
```
Comment (mtmail) [inline lib/SearchDescription.php:530]: boolean variables should start with b, eg. `$bSomething = false`

File: lib/SearchDescription.php [hunk 3]
```diff
@@ -860,14 +865,16 @@ private function queryPoiByOperator(&$oDB, $aParentIDs, $iLimit)
         if ($this->iOperator == Operator::TYPE || $this->iOperator == Operator::NEAR) {
             $sClassTable = $this->poiTable();
             $bCacheTable = $oDB->tableExists($sClassTable);
+            $sIndexName=$this->poiIndex();
```
Comment (mtmail) [inline lib/SearchDescription.php:868]: I'm surprised PHPCS didn't enforce spaces around the = for consistency

File: lib/SearchDescription.php [hunk 4]
```diff
@@ -923,7 +930,8 @@ private function queryPoiByOperator(&$oDB, $aParentIDs, $iLimit)
                         $sSQL .= " WHERE ST_Contains('$sPlaceGeom', l.centroid)";
                     }
 
-                    $sSQL .= $this->oContext->excludeSQL(' AND l.place_id');
+//                    $sSQL .= $this->oContext->excludeSQL(' AND l.place_id');
```
Comment (mtmail) [inline lib/SearchDescription.php:933]: remove commented out code

File: lib/template/includes/introduction.php [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<h2>Welcome to Nominatim</h2>
+<h2>Welcome to Nominatim DEVELOPMENT </h2>
```
Comment (mtmail) [inline lib/template/includes/introduction.php:1]: unrelated change

File: lib/SearchDescription.php [hunk 6]
```diff
@@ -522,26 +523,30 @@ private function queryNearbyPoi(&$oDB, $iLimit)
 
         $aDBResults = array();
         $sPoiTable = $this->poiTable();
+        $sIndexName=$this->poiIndex();
+        $bIndex = $oDB->indexExists($sIndexName);
```
Comment (mtmail) [inline lib/SearchDescription.php:527]: Checking if an index exists during every search is slow. Better to check once (`$oDB->getListOfIndices()`), e.g. in a __construct, and cache the result.

Comment (lonvia) [reply lib/SearchDescription.php:527]: Unfortunately, we can't do this kind of caching because PHP is stateless. So, you'd recreate the list of indexes on every request. 

Looking forward to switching to Python exactly for those kind of reasons.

Comment (nspaeth) [reply lib/SearchDescription.php:527]: Is Switching to Python on a visible roadmap somewhere?

**Comment by lonvia** [comment]
Closing, as this turned out to be unfeasible for reasons mentioned in https://github.com/osm-search/Nominatim/issues/1652#issuecomment-700735875.
